apps/web/src/components/Basenames/UsernameProfileSectionFrames/AddFrameModal.tsx

```diff
@@ -0,0 +1,284 @@
+import { FrameUI } from '@frames.js/render/ui';
+import { useFrame } from '@frames.js/render/use-frame';
+import { useFrameContext } from 'apps/web/src/components/Basenames/UsernameProfileSectionFrames/Context';
+import { theme } from 'apps/web/src/components/Basenames/UsernameProfileSectionFrames/FrameTheme';
+import { SuggestionCard } from 'apps/web/src/components/Basenames/UsernameProfileSectionFrames/SuggestionCard';
+import { Button, ButtonSizes, ButtonVariants } from 'apps/web/src/components/Button/Button';
+import Input from 'apps/web/src/components/Input';
+import Modal, { ModalSizes } from 'apps/web/src/components/Modal';
+import { StaticImageData } from 'next/image';
+import { ChangeEvent, useCallback, useState } from 'react';
+import currencies from './ui/currencies.svg';
+import email from './ui/email.svg';
+import { ShoppingCartIcon } from '@heroicons/react/24/solid';
+import nftProduct from './ui/nftProduct.svg';
+import payouts from './ui/payouts.svg';
+import swap from './ui/swap.svg';
+import starActive from './ui/starActive.svg';
+import { useAccount } from 'wagmi';
+
+export default function AddFrameModal() {
+  const { address } = useAccount();
+  const [frameUrl, setFrameUrl] = useState('');
+  const [farcasterUsername, setFarcasterUsername] = useState('');
+  const handleFarcasterUsernameChange = useCallback((e: ChangeEvent<HTMLInputElement>) => {
+    setFarcasterUsername(e.target.value);
```

Comment:
do we want some validation here? in theory they could put some fraudulent / large string that will be used in `setFrameUrl`